shared/src/vyxal/RegisterHelpers.scala

```diff
@@ -0,0 +1,57 @@
+package vyxal
+
+import vyxal.conversions.{*, given}
+
+import scala.collection.mutable.ArrayBuffer
+
+
+object RegisterHelpers:
+
+	def register(using ctx: Context): ArrayBuffer[VAny] = ctx.globals.register
+	def clear(using Context) = register.length == 0
+	def length(using Context) = VNum(register.length)
+
+	def isEmpty(using Context): Boolean = register.length == 0
+
+	def reverseRegister(using ctx: Context): Unit = 
+		val rev = register.reverse
+		for i <- 0 to register.length - 1 do
+			register.update(i, rev(i))
+		
+	/** Apply a monadic function to an item in the register */	
+	def applyFn(fn: VFun, idx: VNum)(using ctx: Context): Unit =
+		if register.last.isInstanceOf[VPhysical]
+		then
+			val c = Interpreter.executeFn(fn, args = Seq(register(idx.toInt)))
+			register.update(idx.toInt, c)
+
+
+	/** Apply a function to all elements in the register */
+	def map(fn: VFun)(using ctx: Context): Unit =
+		for x <- 0 until register.length do
+			applyFn(fn, VNum(x)) 
```

Comment:
This is now applying the function to indices, not elements. Did you mean
```suggestion
		for x <- register do
			applyFn(fn, x)
```
?

Reply:
applyFn takes an an index

Reply:
Oh, I'm such an idiot